src/main/java/bio/terra/service/JobService.java

```diff
@@ -1,43 +1,74 @@
 package bio.terra.service;
 
+import bio.terra.controller.AuthenticatedUser;
+import bio.terra.controller.RepositoryApiController;
 import bio.terra.model.JobModel;
 import bio.terra.service.exception.InvalidResultStateException;
 import bio.terra.service.exception.JobNotCompleteException;
 import bio.terra.service.exception.JobResponseException;
+import bio.terra.stairway.Flight;
 import bio.terra.stairway.FlightMap;
 import bio.terra.stairway.FlightState;
 import bio.terra.stairway.FlightStatus;
 import bio.terra.stairway.Stairway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
 
 import java.util.ArrayList;
 import java.util.List;
 
 @Component
 public class JobService {
-    // Little class to return HttpStatus when retrieving job result
-    private static class HttpStatusContainer {
-        private HttpStatus statusCode;
-
-        public HttpStatus getStatusCode() {
-            return statusCode;
-        }
-
-        public void setStatusCode(HttpStatus statusCode) {
-            this.statusCode = statusCode;
-        }
-    }
 
+    private static final Logger logger = LoggerFactory.getLogger(JobService.class);
     private final Stairway stairway;
 
     @Autowired
     public JobService(Stairway stairway) {
         this.stairway = stairway;
     }
 
+    private String createJobId() {
+        // in the future, if we have multiple stairways, we may need to maintain a connection from job id to flight id
+        return stairway.createFlightId().toString();
+    }
+
+    public String submit(
+        String description, Class<? extends Flight> flightClass, Object request, AuthenticatedUser userInfo) {
+        String jobId = createJobId();
+        submitToStairway(jobId, flightClass, new FlightMap(description, request), userInfo);
```

Comment:
I do not think FlightMap should know about specific key names. As above, that makes an outbound dependency from stairway.